test/state/state.cpp

```diff
@@ -75,11 +91,32 @@ evmc_message build_message(const Transaction& tx, int64_t execution_gas_limit) n
 /// Validates transaction and computes its execution gas limit (the amount of gas provided to EVM).
 /// @return  Execution gas limit or transaction validation error.
 std::variant<int64_t, std::error_code> validate_transaction(const Account& sender_acc,
-    const BlockInfo& block, const Transaction& tx, evmc_revision rev,
-    int64_t block_gas_left) noexcept
+    const BlockInfo& block, const Transaction& tx, evmc_revision rev, int64_t block_gas_left,
+    int64_t blob_gas_left) noexcept
 {
+    const auto blob_gas_price = fake_exponential(1, block.excess_blob_gas, 3338477);
+
     switch (tx.type)
     {
+    case Transaction::Type::blob:
+        if (rev < EVMC_CANCUN)
+            return make_error_code(TX_TYPE_NOT_SUPPORTED);
+        if (!tx.to.has_value())
+            return make_error_code(CREATE_BLOB_TX);
+        if (tx.blob_hashes.empty())
+            return make_error_code(EMPTY_BLOB_HASHES_LIST);
+        if (tx.blob_hashes.size() > 6)
```

Comment:
I understand that this is a limit for number of blob hashes included in a block, and transaction validity doesn't include this?

Reply:
Yeah. IMO it's already handled by the condition but maybe @chfast wants to take a look also. I don't remember why this additional `BLOB_HASHES_LIST_SIZE_LIMIT_EXCEEDED` error was added. Maybe it's not needed any longer. I would remove it now.

Reply:
If these checks are equivalent we can remove the `tx.blob_hashes.size() > 6`. I only recall discussions about having 3/6 or 2/4 blobs in the block. I'm not sure if blob gas cost handles this exact limit precisely and is enough.

Reply:
Anyway, if the EIP does not mention 6 at all we proceed without it and check against tests.

Reply:
`6*0x20000 == 786432` which is precisely `MAX_BLOB_GAS_PER_BLOCK`